src/adapters/net.cpp

```diff
@@ -161,6 +162,11 @@ namespace net {
         }
       }
 
+      m_status.mac = string_util::trim(file_util::contents(NET_PATH + m_interface + "/address"), isspace);
```

Comment:
This code can be outside of the loop, it doesn't depend on anything in the loop and we may just read the address multiple times for no reason.

Reply:
If the code isn't in the loop then the mac address won't update if it changes. So if the code doesn't belong in the loop then where should it be?

Reply:
Put it outside of the loop. For example at the beginning of the function. The `query` function is called in every update cycle, so it will get updated.

Reply:
Ah, I see what you mean I was confused and thought you meant to put it outside the query function.